Drivers/NER/src/can.c

```diff
@@ -316,7 +316,22 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef *hcan)
     HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO0, rx_header, new_msg.data);
     new_msg.len = rx_header->DLC;
     new_msg.id = rx_header->StdId;
-    enqueue(can1_incoming, new_msg);
+    switch(new_msg.id)
```

Comment:
This still feels icky to have here, feels like we are coupling this too much.

I think what I suggested to you was potentially defining a "can_callback" struct that contained a function pointer and a list of CAN ID's that are relevant to be able to dynamically route callbacks rather than hardcoding

Reply:
Also, the callback for updating shouldn't be made on a driver level IMO. That should be something that is defined at the initialization of the MC monitor

Reply:
I agree with defining a can callback struct, the only reason I didn't do it that way is because I don't know how.

Reply:
I think it would be like
```
typedef struct {
	void (*handler)(can_msg_t); // This is set to a function address &func
	uint8_t *ids; // This can be set to an array of CAN ID's
	uint8_t num_ids; // This can be set to the number of ids in the list
} can_handle_t;
```

Reply:
And then you can maintain a list of the can handlers locally via a linked list OR you can weakly define a list and then have the user override the list with their own array of callbacks. There's a few ways to do it

Reply:
IMO this should be implemented on a higher level when we are slapping shit out of the queue. The only reasons we should implement callbacks on the driver level is if we have like a standard "FAULT" CAN message that every node can send out, but I don't think we plan to do that